pykoopman/koopman.py

```diff
@@ -355,52 +383,60 @@ def koopman_matrix(self):
     @property
     def state_transition_matrix(self):
         """
-        The state transition matrix A satisfies x' = Ax + Bu.
+        The state transition matrix A satisfies y' = Ay or y' = Ay + Bu, respectively,
+        where y = g(x).
         # TODO: consider whether we want to match sklearn and have A and B satisfy
-        # x' = xA + uB instead
+        # y' = yA + uB instead
         """
         check_is_fitted(self, "model")
-        if not isinstance(self.regressor, DMDc):
+        if isinstance(self.regressor, DMDBase):
             raise ValueError(
-                "self.regressor is not DMDc, so object has no "
-                "state_transition_matrix"
+                "this type of self.regressor has no state_transition_matrix"
             )
-        return self.model.steps[-1][1].state_matrix_
+
+        mat = self.koopman_matrix
+        if hasattr(self.model.steps[-1][1], "state_matrix_"):
+            mat = self.model.steps[-1][1].state_matrix_
+        return mat
 
     @property
     def control_matrix(self):
         """
-        The control matrix (or vector) B satisfies x' = Ax + Bu.
+        The control matrix (or vector) B satisfies y' = Ay + Bu.
         """
         check_is_fitted(self, "model")
-        if not isinstance(self.regressor, DMDc):
-            raise ValueError(
-                "self.regressor is not DMDc, so object has no control_matrix"
-            )
+        if isinstance(self.regressor, DMDBase):
+            raise ValueError("this type of self.regressor has no control_matrix")
         return self.model.steps[-1][1].control_matrix_
 
+    def measurement_matrix(self):
+        """
+        The measurement matrix (or vector) C satisfies x = Cy
+        """
+        check_is_fitted(self, "model")
+        if isinstance(self.regressor, DMDBase):
+            raise ValueError("this type of self.regressor has no measurement_matrix")
+        return self.model.steps[0][1].measurement_matrix_
+
     @property
     def projection_matrix(self):
         """
-        The control matrix (or vector) B satisfies x' = Ax + Bu.
+        Projection matrix
         """
```

Comment:
what is projection matrix? and what is output projection matrix?

Reply:
This is just a correction of the doc, which was wrong. The projection matrix U is obtained from a SVD, e.g. Let's say X, X' is high-dimensional spatial data, then A = X' pinv(X) is really expensive and large. So instead one would project onto , e.g., the PCA modes using U, S, V = svd(X), so that Atilde = U.T X' V inv(S) and Atilde = U.T A U